hnn_core/tests/test_batch_simulate.py

```diff
@@ -76,24 +80,126 @@ def test_run_single_sim(batch_simulate_instance):
+
+    with pytest.raises(ValueError, match="Invalid value for the 'backend'"):
+        batch_simulate_instance.simulate_batch(param_combinations,
+                                               backend='invalid')
+
+    with pytest.raises(TypeError, match='verbose must be'):
+        batch_simulate_instance.simulate_batch(param_combinations,
+                                               verbose='invalid')
+
 
 def test_run(batch_simulate_instance, param_grid):
     """Test the run method of the batch_simulate_instance."""
-    results = batch_simulate_instance.run(param_grid, n_jobs=2,
-                                          return_output=True,
-                                          combinations=False)
-
-    assert results is not None
-    assert isinstance(results, list)
-    assert len(results) == len(
+    results_without_cache = batch_simulate_instance.run(param_grid,
+                                                        n_jobs=2,
+                                                        return_output=True,
+                                                        combinations=False,
+                                                        backend='loky',
+                                                        clear_cache=False)
+
+    total_combinations = len(
         batch_simulate_instance._generate_param_combinations(
             param_grid, combinations=False))
+
+    assert results_without_cache is not None
+    assert isinstance(results_without_cache, list)
+    assert len(results_without_cache) == total_combinations
+
+    results_with_cache = batch_simulate_instance.run(param_grid,
+                                                     n_jobs=2,
+                                                     return_output=True,
+                                                     combinations=False,
+                                                     backend='loky',
+                                                     clear_cache=True)
+
+    assert results_with_cache is not None
+    assert isinstance(results_with_cache, list)
+    assert len(results_with_cache) == 0
+
+    # Validation Tests
+    with pytest.raises(TypeError, match='param_grid must be'):
+        batch_simulate_instance.run('invalid_param_grid')
+
+    with pytest.raises(TypeError, match='n_jobs must be'):
+        batch_simulate_instance.run(param_grid, n_jobs='invalid')
+
+    with pytest.raises(ValueError, match="Invalid value for the 'backend'"):
+        batch_simulate_instance.run(param_grid, backend='invalid_backend')
+
+    with pytest.raises(TypeError, match='verbose must be'):
+        batch_simulate_instance.run(param_grid, verbose='invalid')
+
+    with pytest.raises(TypeError, match='clear_cache must be'):
+        batch_simulate_instance.run(param_grid, clear_cache='invalid')
+
+    # Callables Test
+    batch_simulate_instance.summary_func = lambda x: x
+    assert callable(batch_simulate_instance.summary_func)
```

Comment:
Sorry I should have been more explicit here, it's best to test that an error is raised for type checks

Reply:
Sorry, my mistake! I forgot to mention the test that caused an error